.github/workflows/build-test.yml

```diff
@@ -130,18 +130,47 @@ jobs:
                 TARGET_DOCKER_TEST_IMAGE: ${{ matrix.image }}
               run: ./ci/test.sh
 
-    test-mac-optional:
+    test-mac-azure-optional:
```

Comment:
why cannot we just run tests with and don't duplicate build configuration?
```
matrix:
  cloud: [ 'AZURE', 'GCP' ]
  nodeVersion: ['16.x', '18.x']
```

can we take it for windows and mac?

Reply:
and let's add `fail-fast: false` to not cancel other builds when one matrix item fails